src/Utilities/TypeTraits.hpp

```diff
@@ -906,6 +906,39 @@ using is_callable_t = typename is_callable<T, Args...>::type;
   using is_##METHOD_NAME##_callable_t =                                        \
       is_##METHOD_NAME##_callable_r_t<AnyReturnType##METHOD_NAME, T, Args...>;
 
+/*!
+ * \ingroup TypeTraitsGroup
+ * \brief Generate a type trait to check if a class has a `static constexpr`
+ * variable, optionally also checking its type.
+ *
+ * \example
+ * \snippet Utilities/Test_TypeTraits.cpp CREATE_HAS_EXAMPLE
+ *
+ * \see `CREATE_IS_CALLABLE`
+ */
+#define CREATE_HAS(CONSTEXPR_NAME)                                           \
+  struct NoSuchType##CONSTEXPR_NAME {};                                      \
+                                                                             \
+  template <typename CheckingType,                                           \
+            typename VariableType = NoSuchType##CONSTEXPR_NAME,              \
+            typename = cpp17::void_t<>>                                      \
+  struct has_##CONSTEXPR_NAME : std::false_type {};                          \
+                                                                             \
+  template <typename CheckingType, typename VariableType>                    \
+  struct has_##CONSTEXPR_NAME<CheckingType, VariableType,                    \
+                              cpp17::void_t<std::remove_const_t<decltype(    \
+                                  CheckingType::CONSTEXPR_NAME)>>>           \
+      : cpp17::disjunction<                                                  \
+            std::is_same<VariableType, NoSuchType##CONSTEXPR_NAME>,          \
+            std::is_same<                                                    \
+                std::remove_const_t<decltype(CheckingType::CONSTEXPR_NAME)>, \
+                VariableType>> {};                                           \
+                                                                             \
+  template <typename CheckingType,                                           \
+            typename VariableType = NoSuchType##CONSTEXPR_NAME>              \
```

Comment:
`NoSuchType*****` (if someone has that many pointers to a thing that isn't useful, it's their fault...)

Reply:
This is not the standard `NoSuchType` but it has the `CONSTEXPR_NAME` appended (defined a few lines up). Should I add pointers nonetheless?

Reply:
Sorry, I'm suggesting using the standard `NoSuchType` and just adding a bunch of pointers. The reason is that adding the pointers results in fewer template classes being created and reducing compile time and memory consumption. Does that make sense?